zyte_spider_templates/spiders/serp.py

```diff
@@ -48,6 +58,83 @@ class SerpMaxPagesParam(BaseModel):
     )
 
 
+@document_enum
+class SerpItemType(str, Enum):
+    article: str = "article"
+    """
+    Article data from result URLs.
+    """
+
+    articleList: str = "articleList"
+    """
+    Article list data from result URLs.
+    """
+
+    # forumThread: str = "forumThread"
+    """
+    Thread data from result URLs.
+    """
+
+    jobPosting: str = "jobPosting"
+    """
+    Job posting data from result URLs.
+    """
+
+    product: str = "product"
+    """
+    Product data from result URLs.
+    """
+
+    productList: str = "productList"
+    """
+    Product list data from result URLs.
+    """
+
+    serp: str = "serp"
+    """
+    Search engine results page data.
+    """
+
+
+# NOTE: serp is excluded on purposed, since it is not used below.
+# TODO: Add a test to make sure that this is in sync with the enum class above.
+ITEM_TYPE_CLASSES = {
+    SerpItemType.article: Article,
+    SerpItemType.articleList: ArticleList,
+    # SerpItemType.forumThread: ForumThread,
+    SerpItemType.jobPosting: JobPosting,
+    SerpItemType.product: Product,
+    SerpItemType.productList: ProductList,
+}
+
+
+class SerpItemTypeParam(BaseModel):
+    item_type: SerpItemType = Field(
+        title="Item type",
+        description="Data type of the output items.",
+        default=SerpItemType.serp,
+        json_schema_extra={
+            "enumMeta": {
+                # TODO: Add a test to make sure this is in sync with the enum class above.
+                # TODO: Try automating the generation of this metadata from the enum type above.
+                SerpItemType.serp: {
```

Comment:
Hm, that's an interesting idea. I was thinking that the default here would be None, which means "don't go to the results, output the links".